state/upgrades.go

```diff
@@ -3482,3 +3482,31 @@ func TranslateK8sServiceTypes(pool *StatePool) error {
 	}
 	return nil
 }
+
+// UpdateDHCPAddressConfigs ensures that any addresses in the ip.addresses
+// collection with the removed "dynamic" address configuration method are
+// updated to indicate the "dhcp" configuration method.
+func UpdateDHCPAddressConfigs(pool *StatePool) error {
+	st := pool.SystemState()
+
+	col, closer := st.db().GetRawCollection(ipAddressesC)
+	defer closer()
+
+	iter := col.Find(bson.M{"config-method": "dynamic"}).Iter()
+
+	var ops []txn.Op
+	var doc bson.M
+	for iter.Next(&doc) {
+		ops = append(ops, txn.Op{
+			C:      ipAddressesC,
+			Id:     doc["_id"],
+			Assert: txn.DocExists,
+			Update: bson.M{"$set": bson.M{"config-method": network.ConfigDHCP}},
+		})
+	}
+
+	if len(ops) > 0 {
```

Comment:
You probably need to add a:

```
if err := iter.Close(); err != nil {
  return errors.Trace(err)
}
```

before checking `len(ops)`.

Reply:
Good catch; ta.